plugins/obs-ffmpeg/ffmpeg-encoded-output.c

```diff
@@ -0,0 +1,905 @@
+	config.video_bitrate = encoder_bitrate(vencoder);
+	config.audio_bitrate = obs_encoder_get_sample_rate(aencoder) / 1000;
+	config.gop_size = 250;
+	config.video_encoder = "";
+	config.video_encoder_id = AV_CODEC_ID_H264;
+	config.audio_encoder = "";
+	config.audio_encoder_id = AV_CODEC_ID_AAC;
+	config.video_settings = 0;
+	config.audio_settings = 0;
+	config.scale_width = 0;
+	config.scale_height = 0;
+	config.width = obs_encoder_get_width(vencoder);
+	config.height = obs_encoder_get_height(vencoder);
+	config.audio_tracks = audio_output_get_channels(audio);
+	config.audio_mix_count = get_audio_mix_count(config.audio_tracks);
+	config.format = obs_to_ffmpeg_video_format(
+			video_output_get_format(video));
+
+	if (format_is_yuv(voi->format)) {
+		config.color_range =
+				voi->range == VIDEO_RANGE_FULL ?
+						AVCOL_RANGE_JPEG :
+						AVCOL_RANGE_MPEG;
+		config.color_space =
+				voi->colorspace == VIDEO_CS_709 ?
+						AVCOL_SPC_BT709 :
+						AVCOL_SPC_BT470BG;
+	} else {
+		config.color_range = AVCOL_RANGE_UNSPECIFIED;
+		config.color_space = AVCOL_SPC_RGB;
+	}
+
+	if (config.format == AV_PIX_FMT_NONE) {
+		blog(LOG_DEBUG, "invalid pixel format used for FFmpeg output");
+		return false;
+	}
+
+	if (!config.scale_width)
+		config.scale_width = config.width;
+	if (!config.scale_height)
+		config.scale_height = config.height;
+
+	success = ffmpeg_data_init(&stream->ff_data, &config);
+	if (!success)
+		return -1;
+
+	return ret;
+}
+
+static int proto_connect_time(struct ffmpeg_encoded_output *stream)
+{
+	UNUSED_PARAMETER(stream);
+
+	return 0;
+}
+
+static void proto_close(struct ffmpeg_encoded_output *stream)
+{
+	ffmpeg_data_free(&stream->ff_data);
+}
+
+static inline int64_t _rescale_ts(struct ffmpeg_encoded_output *stream,
+		int64_t val, int idx)
+{
+	AVStream *avstream = stream->ff_data.output->streams[idx];
+
+	return av_rescale_q_rnd(val / avstream->codec->time_base.num,
+			avstream->codec->time_base, avstream->time_base,
+			AV_ROUND_NEAR_INF | AV_ROUND_PASS_MINMAX);
+}
+
+static int proto_send_packet(struct ffmpeg_encoded_output *stream,
+		struct encoder_packet *obs_packet, bool is_header)
+{
+	AVPacket av_packet = { 0 };
+	int ret = 0;
+	int streamIdx = (obs_packet->type == OBS_ENCODER_VIDEO) ? 0 : 1;
+
+	//2. send
+	av_init_packet(&av_packet);
+
+	av_packet.data = obs_packet->data;
+	av_packet.size = (int) obs_packet->size;
+	av_packet.stream_index = streamIdx;
+	av_packet.pts = _rescale_ts(stream, obs_packet->pts, streamIdx);
+	av_packet.dts = _rescale_ts(stream, obs_packet->dts, streamIdx);
+
+	if (obs_packet->keyframe)
+		av_packet.flags = AV_PKT_FLAG_KEY;
+
+	ret = av_interleaved_write_frame(stream->ff_data.output, &av_packet);
+
+	if (is_header)
+		bfree(obs_packet->data);
+	else
+		obs_encoder_packet_release(obs_packet);
+
+	stream->total_bytes_sent += obs_packet->size;
+
+	return ret;
+
+}
+
+static void proto_set_output_error(struct ffmpeg_encoded_output *stream)
+{
+	UNUSED_PARAMETER(stream);
+}
```

Comment:
can we retrieve errors from  libsrt through avformat ? or would that require direct implementation ?

Reply:
No, it doesn't look like we can do that.